src/Steam.Common/SteamId.cs

```diff
@@ -0,0 +1,492 @@
+﻿using System;
+using System.Text.RegularExpressions;
+
+namespace Steam
+{
+    /// <summary>
+    /// A globally unique identifier for Steam accounts, groups, lobbies, and chat rooms
+    /// </summary>
+    public readonly struct SteamId : IEquatable<SteamId>
+    {
+        private const uint AccountIdMask = 0xFFFFFFFF;
+
+        private const uint AccountInstanceMask = 0xFFFFF;
+        private const int AccountInstanceShift = 32;
+
+        private const uint AccountTypeMask = 0xF;
+        private const int AccountTypeShift = 52;
+
+        private const int AccountUniverseShift = 56;
+
+        private const int ChatAccountInstanceMask = 0xFFF;
+
+        private readonly static Regex Steam2Regex = new Regex(@"STEAM_([0-4]):(0|1):(\d+)");
```

Comment:
The order of these modifiers is inconsistent with the rest. We should pick one way and stick with it.

Reply:
I'd say `static` should be followed by `readonly`. I can't figure out any way to enforce this however.